pinot-broker/src/main/java/com/linkedin/pinot/broker/broker/BrokerServerBuilder.java

```diff
@@ -198,23 +202,35 @@ public void buildHTTP() {
     Configuration c = _config.subset(CLIENT_CONFIG_PREFIX);
     BrokerClientConf clientConfig = new BrokerClientConf(c);
 
-    _server = new Server(clientConfig.getQueryPort());
+    Preconditions.checkArgument(clientConfig.getQueryPort() > 0);
+    URI baseUri = URI.create("http://0.0.0.0:" + Integer.toString(clientConfig.getQueryPort()) + "/");
 
-    WebAppContext context = new WebAppContext();
-    context.addServlet(PinotClientRequestServlet.class, "/query");
-    context.addServlet(PinotBrokerHealthCheckServlet.class, "/health");
-    context.addServlet(PinotBrokerRoutingTableDebugServlet.class, "/debug/routingTable/*");
-    context.addServlet(PinotBrokerTimeBoundaryDebugServlet.class, "/debug/timeBoundary/*");
+    _httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri);
 
+    WebappContext context;
     if (clientConfig.enableConsole()) {
-      context.setResourceBase(clientConfig.getConsoleWebappPath());
+      // Use "" for context path to make it "/"
+      context = new WebappContext("brokerServerContext", "", clientConfig.getConsoleWebappPath());
+
     } else {
-      context.setResourceBase("");
+      context = new WebappContext("brokerServerContext");
     }
+    ServletRegistration servletRegistration;
```

Comment:
dont use servlets. see controller example.

Reply:
I will work on porting to Jersey for the next release.

Reply:
ok